criu/uffd.c

```diff
@@ -262,14 +262,18 @@ static int uffd_api_ioctl(void *arg, int fd, pid_t pid)
 	return ioctl(fd, UFFDIO_API, uffdio_api);
 }
 
-int uffd_open(int flags, unsigned long *features)
+int uffd_open(int flags, unsigned long *features, bool called_from_kerndat)
 {
 	struct uffdio_api uffdio_api = { 0 };
 	int uffd;
 
 	uffd = syscall(SYS_userfaultfd, flags);
 	if (uffd == -1) {
-		pr_perror("Lazy pages are not available");
+		if (!called_from_kerndat) {
```

Comment:
You're focusing too much on random stuff happening around instead of focusing on high-level intent. What if somebody writes a new usage of this function from outside of kerndat but they don't care if it succeeds? Better names include `userfaultfd_should_succeed`, `userfaultfd_ok_to_fail`, `userfaultfd_print_error`.

Reply:
Got it! Looks like I should be emphasizing the function of the variable rather than where it's being called from.